compiler/p/code

```diff
@@ -0,0 +1,3682 @@
+         window++;
+         }
+      }
+   }
+
+bool OMR::Power::CodeGenerator::checkAndFetchRequestor(TR::Instruction *instr, TR::Instruction **q)
+   {
+   if (_constantData)
+      {
+      return _constantData->getRequestorsFromNibble(instr, q, true);
+      }
+   return false;
+   }
+
+// PPC Codegen Peephole Pass
+// one forward pass over the code looking for a variety of peephole
+// opportunities
+void OMR::Power::CodeGenerator::doPeephole()
+   {
+   static char *disablePeephole = feGetEnv("TR_DisablePeephole");
+   if (disablePeephole)
+      return;
+
+   int syncPeepholeWindow = self()->comp()->isOptServer() ? 12 : 6;
+
+   if (self()->comp()->getOptLevel() == noOpt)
+      return;
+
+   TR::Instruction *instructionCursor = self()->getFirstInstruction();
+
+   while (instructionCursor)
+      {
+      self()->setCurrentBlockIndex(instructionCursor->getBlockIndex());
+
+      if ((TR::Compiler->target.cpu.id() == TR_PPCp6) && instructionCursor->isTrap())
+         {
+#if defined(AIXPPC)
+         trapPeephole(self(),instructionCursor);
+#endif
+         }
+      else
+         {
+         switch(instructionCursor->getOpCodeValue())
+            {
+            case TR::InstOpCode::stw:
+#if defined(TR_HOST_64BIT)
+            case TR::InstOpCode::std:
+#endif
+               {
+               lhsPeephole(self(), instructionCursor);
+               break;
+               }
+            case TR::InstOpCode::mr:
+               {
+               mrPeepholes(self(), instructionCursor);
+               break;
+               }
+            case TR::InstOpCode::cmpi4:
+               {
+               if (TR::Compiler->target.is32Bit())
+                  recordFormPeephole(self(), instructionCursor);
+               break;
+               }
+            case TR::InstOpCode::cmpi8:
+               {
+               if (TR::Compiler->target.is64Bit())
+                  recordFormPeephole(self(), instructionCursor);
+               break;
+               }
+            case TR::InstOpCode::sync:
+               {
+               syncPeephole(self(), instructionCursor, syncPeepholeWindow);
+               break;
+               }
+            case TR::InstOpCode::lwsync:
+               {
+               syncPeephole(self(), instructionCursor, syncPeepholeWindow);
+               break;
+               }
+            case TR::InstOpCode::isync:
+               {
+               syncPeephole(self(), instructionCursor, syncPeepholeWindow);
+               break;
+               }
+            default:
+               {
+               wawPeephole(self(), instructionCursor);
+               break;
+               }
+            }
+         }
+      instructionCursor = instructionCursor->getNext();
+      }
+   }
+// end of PPC Codegen Peephole routines
+
+bool OMR::Power::CodeGenerator::supportsAESInstructions()
+   {
+    if ( TR::Compiler->target.cpu.getPPCSupportsAES() && !self()->comp()->getOption(TR_DisableAESInHardware))
+      ret
```

Comment:
This looks like you accidentally copied OMRCodeGenerator.cpp to this compiler/p/code file?